src/account/operations/output_claiming.rs

```diff
@@ -225,93 +230,82 @@ impl AccountHandle {
-                        if let Some(sdr) = unlock_conditions.storage_deposit_return() {
-                            storage_deposit = true;
-                            return_amount = sdr.amount();
-                            // create return output
-                            outputs_to_send.push(
-                                BasicOutputBuilder::new_with_amount(sdr.amount())?
-                                    .add_unlock_condition(UnlockCondition::Address(AddressUnlockCondition::new(
-                                        *sdr.return_address(),
-                                    )))
-                                    .finish_output()?,
-                            );
-                        }
-                    }
-                    if storage_deposit {
-                        // for own output subtract the return amount
-                        new_amount += output_data.output.amount() - return_amount;
-                        if let Some(native_tokens) = output_data.output.native_tokens() {
-                            new_native_tokens.add_native_tokens(native_tokens.clone())?;
-                        }
-                    } else {
-                        new_amount += output_data.output.amount();
-                        if let Some(native_tokens) = output_data.output.native_tokens() {
-                            new_native_tokens.add_native_tokens(native_tokens.clone())?;
-                        }
+                    new_amount += output_data.output.amount();
+                    if let Some(native_tokens) = output_data.output.native_tokens() {
+                        new_native_tokens.add_native_tokens(native_tokens.clone())?;
                     }
                 }
             }
 
-            // Check if the new amount is enough for the storage deposit, otherwise increase it to this
             let option_native_token = if new_native_tokens.is_empty() {
                 None
             } else {
                 Some(new_native_tokens.clone().finish()?)
             };
+
+            // Check if the new amount is enough for the storage deposit, otherwise increase it to this
             let required_storage_deposit = minimum_storage_deposit(
                 &byte_cost_config,
                 &first_account_address.address.inner,
                 &option_native_token,
             )?;
-            let mut additional_inputs = Vec::new();
 
+            let mut additional_inputs = Vec::new();
             if new_amount < required_storage_deposit {
                 // add more inputs
                 for output_data in &possible_additional_inputs {
                     // Recalculate every time, because new intputs can also add more native tokens, which would increase
                     // the storage deposit cost
-                    let required_storage_deposit = minimum_storage_deposit(
+                    let mut required_storage_deposit = minimum_storage_deposit(
```

Comment:
But maybe here it should be like
```suggestion
                    required_storage_deposit = minimum_storage_deposit(
```
and the first declaration should be mutable

Reply:
fixed